common/client/node_lifecycle.go

```diff
@@ -168,10 +168,8 @@ func (n *node[CHAIN_ID, HEAD, RPC]) aliveLoop() {
 				n.declareUnreachable()
 				return
 			}
-			_, latestChainInfo := n.StateAndLatest()
-			if outOfSync, liveNodes := n.isOutOfSyncWithPool(latestChainInfo); outOfSync {
+			if outOfSync, liveNodes := n.isOutOfSyncWithPool(); outOfSync {
```

Comment:
Out of curiosity, why bundle the bug fix to the OutOfSync logic here with the Transaction Sender fixes below? They seem unrelated and could have been done as separate PRs. It also makes the individual PRs smaller and easier to review.